Tests/src/PublicAPITests/HotSpotSequenceSystemTests.cpp

```diff
@@ -0,0 +1,648 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Awaitable.h"
+#include "CSP/CSPFoundation.h"
+#include "CSP/Systems/HotSpotSequence/HotSpotSequenceSystem.h"
+#include "CSP/Systems/Spaces/SpaceSystem.h"
+#include "CSP/Systems/SystemsManager.h"
+#include "SpaceSystemTestHelpers.h"
+#include "TestHelpers.h"
+#include "UserSystemTestHelpers.h"
+
+#include <iostream>
+
+namespace
+{
+
+bool RequestPredicate(const csp::systems::ResultBase& Result)
+{
+	return Result.GetResultCode() != csp::systems::EResultCode::InProgress;
+}
+
+void WaitForCallback(bool& CallbackCalled)
+{
+	// Wait for message
+	auto Start		 = std::chrono::steady_clock::now();
+	auto Current	 = std::chrono::steady_clock::now();
+	int64_t TestTime = 0;
+
+	while (CallbackCalled == false && TestTime < 20)
+	{
+		std::this_thread::sleep_for(50ms);
+
+		Current	 = std::chrono::steady_clock::now();
+		TestTime = std::chrono::duration_cast<std::chrono::seconds>(Current - Start).count();
+	}
+}
+
+} // namespace
+
+void CreateHotspotgroup(csp::systems::HotSpotSequenceSystem* HotspotSequenceSystem,
+						const csp::common::String& GroupName,
+						const csp::common::Array<csp::common::String>& Items,
+						csp::systems::HotSpotGroup& OutSequence,
+						csp::systems::EResultCode ExpectedResultCode				  = csp::systems::EResultCode::Success,
+						csp::systems::ERequestFailureReason ExpectedResultFailureCode = csp::systems::ERequestFailureReason::None)
+{
+	auto [Result]
+		= Awaitable(&csp::systems::HotSpotSequenceSystem::CreateHotspotGroup, HotspotSequenceSystem, GroupName, Items).Await(RequestPredicate);
+
+	EXPECT_EQ(Result.GetResultCode(), ExpectedResultCode);
+	EXPECT_EQ(Result.GetFailureReason(), ExpectedResultFailureCode);
+
+	if (ExpectedResultCode == csp::systems::EResultCode::Success)
+	{
+		csp::systems::HotSpotGroup group = Result.GetHotSpotGroup();
+
```

Comment:
Check the group name?

Reply:
We check metadata tags now for this

Reply:
What I meant was is it worth confirming the group name we asked for is the one CHS returned? They could, in theory, return a different name.

Reply:
Yes, we check this, as when the group is retrieved, we extract the group name from metadata (as opposed to a constructed unique key)